docs/index.md

```diff
@@ -30,92 +28,159 @@ This means that you can write your entire application in a single language, Rust
+{
+    [SpacetimeDB.PrimaryKey]
+    uint playerId;
+    string name;
+    uint age;
+    Identity user;
+}
+```
+<!-- TODO: switchable language widget.
+```rust
+#[spacetimedb::table(name = players, public)]
+pub struct Player {
+   #[primary_key]
+   id: u64,
+   name: String,
+   age: u32,
+   user: Identity,
+}
+```
+-->
+
+The contents of a table can be read and updated by [reducers](#reducer).
+Tables marked `public` can also be read by [clients](#client).
+
+### Reducer
+A **reducer** is a function exported by a [module](#module).
+Connected [clients](#client-side-sdks) can call reducers to interact with the module.
+This is a form of [remote procedure call](https://en.wikipedia.org/wiki/Remote_procedure_call).
+Reducers can be invoked across languages. For example, a Rust [module](#module) can export a reducer like so:
+
+```csharp
+[SpacetimeDB.Reducer]
+public static void SetPlayerName(ReducerContext ctx, uint playerId, string name)
+{
+    // ...
+}
+```
+<!-- TODO: switchable language widget.
+```rust
+#[spacetimedb::reducer]
+pub fn set_player_name(ctx: &spacetimedb::ReducerContext, id: u64, name: String) -> Result<(), String> {
+   // ...
+}
+```
+-->
+
+And a C# [client](#client) can call that reducer:
+
+```cs
+void Main() {
+   // ...setup code, then...
+   Connection.Reducer.SetPlayerName(57, "Marceline");
+}
+```
+
+These look mostly like regular function calls, but under the hood, the client sends a request over the internet, which the module processes and responds to.
+
+The `ReducerContext` passed into a reducer includes information about the caller's [identity](#identity) and [address](#address).
+It also allows accessing the database and scheduling future operations.
+
+### Client
+A **client** is an application that connects to a [module](#module). A client logs in using an [identity](#identity) and receives an [address](#address) to identify the connection. After that, it can call [reducers](#reducer) and query public [tables](#table).
+
+Clients are written using the [client-side SDKs](#client-side-sdks). These are regular software libraries. Unlike the [server-side libraries](#server-side-libraries), they don't require the use of any special build tools.
+
+### Identity
+
+A SpacetimeDB `Identity` allows someone to log in to a module.
+
+Users receive an `Identity` by logging in through an [OpenID Connect](https://openid.net/developers/how-connect-works/) provider, such as Google or Facebook. Your module can choose which providers it trusts.
+
+A user's `Identity` is attached to every [reducer call](#reducer) they make, and you can use this to decide what they are allowed to do.
+
+Modules themselves also have Identities.  If your module lives on a shared SpacetimeDB [host](#host), like https://testnet.spacetimedb.com, it will automatically be issued an `Identity` to distinguish it from other modules. Your client application will need to provide this `Identity` when connecting to the host.
```

Comment:
- Remove ref to testnet.
- This is true even of non-shared hosts, so don't hedge.